onnxruntime/python/tools/transformers/benchmark.py

```diff
@@ -339,8 +339,40 @@ def get_latency_result(runtimes, batch_size):
     }
 
 
+def inference_ort(ort_session, ort_inputs, result_template, repeat_times, batch_size):
+    result = {}
+    runtimes = timeit.repeat(lambda: ort_session.run(None, ort_inputs), number=1, repeat=repeat_times)
+    result.update(result_template)
+    result.update({"engine": "onnxruntime"})
+    result.update(get_latency_result(runtimes, batch_size))
+    return result
+
+
+def inference_ort_with_io_binding(ort_session, ort_inputs, result_template, repeat_times, ort_output_names,
+                                  last_state_buffer, pooler_buffer, batch_size, sequence_length, hidden_size, device):
+    result = {}
+    # Bind inputs and outputs to onnxruntime session
+    io_binding = ort_session.io_binding()
+    # Bind inputs to device
+    for name in ort_inputs.keys():
+        np_input = torch.from_numpy(ort_inputs[name]).to(device)
+        io_binding.bind_input(name, np_input.device.type, 0, numpy.longlong, np_input.shape, np_input.data_ptr())
+    # Bind outputs buffers with the sizes needed
+    io_binding.bind_output(ort_output_names[0], last_state_buffer.device.type, 0, numpy.float32,
```

Comment:
This might not work for other models (like gpt2).

Shall we use automatic way to detect number of outputs and shape of output tensors? For example, run one inference without IO binding given batch_size and sequence_length. Then we know the shape of each output tensor.

Reply:
Good suggestion. Just one thing, if we do test run, should we just reuse the output arrays, meaning no need to pre-allocate a large buffer?

Reply:
Reuse sounds good to me.